src/utils/Config.ts

```diff
@@ -0,0 +1,64 @@
+import { Maybe } from "../types/utils";
+
+export class Config {
+    private static instance: Config;
+    private constructor() {}
+
+    public static getInstance(): Config {
+        if (!Config.instance) {
+            Config.instance = new Config();
+        }
+        return Config.instance;
+    }
+
+    get mode(): string {
+        return import.meta.env.MODE;
+    }
+
+    get isDevelopment(): boolean {
+        return this.mode === "development";
+    }
+
+    get presentationTitle(): Maybe<string> {
+        return import.meta.env.VITE_PRESENTATION_TITLE;
+    }
+
+    get presentationVariant(): Maybe<string> {
+        return import.meta.env.VITE_PRESENTATION_VARIANT;
+    }
+
+    /**
+     * returns the presentationVariant only if it's a valid AppVariant
+     */
+    get appPresentationVariant(): Maybe<AppVariant> {
+        const variant = this.presentationVariant;
+        return this.isValidAppVariant(variant) ? variant : undefined;
+    }
+
+    get presentationType(): Maybe<PresentationType> {
+        const type = import.meta.env.VITE_PRESENTATION_TYPE;
+        return type === "webapp" || type === "widget" ? type : undefined;
+    }
```

Comment:
As discussed, let's try to remove all the Maybe that make sense to remove. And check in the usage that we don't have default values lingering.